subsys/bluetooth/host/gatt.c

```diff
@@ -2516,7 +2516,7 @@ static void parse_read_by_uuid(struct bt_conn *conn,
 		BT_DBG("handle 0x%04x len %u value %u", handle, rsp->len, len);
 
 		/* Update start_handle */
-		params->by_uuid.start_handle = handle;
+		params->by_uuid.start_handle = handle + 1;
```

Comment:
While this seems to be correct the update should probably be after the function call otherwise it may associate the read with a different handle (+1).